servers/rendering/renderer_rd/forward_clustered/scene_shader_forward_clustered.cpp

```diff
@@ -309,10 +309,11 @@ void SceneShaderForwardClustered::ShaderData::set_code(const String &p_code) {
 						}
 
 						RD::PipelineDepthStencilState depth_stencil = depth_stencil_state;
-						if (depth_pre_pass_enabled && casts_shadows()) {
+						if (depth_pre_pass_enabled && casts_shadows() && !uses_depth_pre_pass) {
```

Comment:
`uses_depth_pre_pass` needs to be renamed imo. Without looking at L117 I would have never guessed that it had anything to do with alpha pre-passes.

Right now this code reads like it always evaluates as false.

Reply:
Yes, especially given the comment and the ``depth_pre_pass_enabled`` bool

Reply:
(I approved for merge, but will wait for those changes first of course.)

Reply:
renamed to ``uses_depth_prepass_alpha``